File: dotnet/src/Core/Models/Messages/AgienceChatMessageArgs.cs
using Microsoft.SemanticKernel;

namespace Agience.Core.Models.Messages
{
    public class AgienceChatMessageArgs
    {
        public string AgencyId { get; set; }
        public ChatMessageContent Message { get; set; }
        public string AgentId { get; set; }
    }
}

File: python/src/core/models/messages/agience_chat_message_args.py
from typing import Optional
from pydantic import BaseModel
from semantic_kernel.contents import ChatMessageContent


class AgienceChatMessageArgs(BaseModel):
    message: Optional[ChatMessageContent] = None
    agent_id: Optional[str] = None

    @property
    def latest_message(self) -> Optional[str]:
        return self.message.content if self.message else None
